### PR TITLE
fix(linter): check for all .eslintrc.* files when generating a package

### PR DESCRIPTION
**Pull request by SimonMo88**
## Current Behavior
- When generating new applications or packages, a `.eslintrc.json` file is still generated if we are using other eslint config file  formats, for example `.eslintrc`.

## Expected Behavior
- If a `.eslintrc.*` file that matches any of the possible `.eslintrc.*` formats listed [here](https://eslint.org/docs/latest/user-guide/configuring/configuration-files#configuration-file-formats), in the root of the monorepo. We should not generate a new `.eslintrc.json` file.

## Related Issue(s)
Enhances https://github.com/nrwl/nx/pull/10080

### CODE REVIEW

**Comment by FrozenPandaz** [comment]
Thank you for adding support for these files! We also have a migration which will add these global files as an input into the `lint` target:

https://github.com/nrwl/nx/blob/master/packages/linter/src/migrations/update-15-0-0/add-eslint-inputs.ts#L17

Could you take a look at adding support for these files there as well?

**Comment by Fried-Chicken** [comment]
Hey @FrozenPandaz

I've added the new eslint files to the migrations now. I hope it's correct, I noticed the snapshot before showed "!{projectRoot}/.eslintrc.json"` in  the `namedInputs.production` array for the `.eslintrc.js` test. Was this intentional?

File: packages/linter/src/migrations/update-15-0-0/add-eslint-inputs.spec.ts [hunk 1]
```diff
@@ -6,135 +6,193 @@ import {
...
-    expect(updated).toMatchInlineSnapshot(`
-      Object {
-        "namedInputs": Object {
-          "default": Array [
-            "{projectRoot}/**/*",
-            "sharedGlobals",
-          ],
-          "production": Array [
-            "default",
-            "!{projectRoot}/.eslintrc.json",
```
Comment (Fried-Chicken) [inline packages/linter/src/migrations/update-15-0-0/add-eslint-inputs.spec.ts:118]: @FrozenPandaz This is the snapshot in question.

File: packages/linter/src/generators/utils/eslint-file.ts [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
+  '.eslintrc.js',
+  '.eslintrc.cjs',
+  '.eslintrc.yaml',
+  '.eslintrc.yml',
+  '.eslintrc.json',
+];
 
 export function findEslintFile(tree: Tree): string | null {
-  for (const file of eslintFileList) {
+  for (const file of eslintConfigFileWhitelist) {
```
Comment (meeroslav) [inline packages/linter/src/generators/utils/eslint-file.ts:13]: Why did you rename this constant?

File: packages/linter/src/migrations/update-15-0-0/add-eslint-inputs.spec.ts (hunk 1 above)
Comment (meeroslav) [reply packages/linter/src/migrations/update-15-0-0/add-eslint-inputs.spec.ts:118]: Probably a typo. Interestingly how that worked.

**Review by meeroslav — CHANGES_REQUESTED** [review]
Left one minor comment.

Everything else looks good to me! Thank you

File: packages/linter/src/generators/utils/eslint-file.ts (hunk 2 above)
Comment (Fried-Chicken) [reply packages/linter/src/generators/utils/eslint-file.ts:13]: Hey @meeroslav

I renamed the constant mainly because I thought it would be a more accurate description of what was as I didn't want to leave anything up to assumption, since I found when I was working on this PR, I often found myself wondering if this variable name was too ambiguous.

I not precious about it though, I can revert this change if you like. Please let me know and I will update it for you 👍 

Thanks for reviewing my PR 🥰

Comment (meeroslav) [reply packages/linter/src/generators/utils/eslint-file.ts:13]: Makes sense. I would probably go with `eslintConfigFileNames` but I guess this is fine as well.

**Review by meeroslav — APPROVED** [review]

File: packages/linter/src/generators/utils/eslint-file.ts (hunk 2 above)
Comment (Fried-Chicken) [reply packages/linter/src/generators/utils/eslint-file.ts:13]: Oh that's an even better suggestion! Maybe you can update it in your next PR, otherwise I can also raise a new PR.
